storage-proofs/src/vde.rs

```diff
@@ -86,25 +86,23 @@ where
     Ok(H::sloth_decode(&key, &node_data, sloth_iter))
 }
 
-pub fn decode_domain_block<'a, H, G>(
-    graph: &'a G,
+pub fn decode_domain_block<'a, H>(
+    degree: usize,
```

Comment:
To be honest, we probably don't need the degree either. It only eventually gets used to assert that we have the right number of parents when performing the kdf. But that actually must be true anyway, if `parents` is calculated correctly. For that reason, I'm tentatively in favor of removing `degree` here — since it's only being passed around to perform a run-time verification which was probably useful when written but doesn't justify contorting the plumbing throughout.

Reply:
> But that actually must be true anyway, if `parents` is calculated correctly.

Why?

Reply:
Because the definitions of the `parents` methods for both `Graph`  and `ZigZagGraph` ensure that the length of the results they return will be equal to the graph degree. If we want an explicit guarantee of this, we could put the assertion in the implemention of `parents`.